tfx_addons/sampler/component.py

```diff
@@ -0,0 +1,106 @@
+# Copyright 2021 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Sampling component definition."""
+
+from typing import List, Optional, Text
+
+from tfx import types
+from tfx.dsl.components.base import base_beam_component, executor_spec
+from tfx.types import channel_utils, standard_artifacts
+from tfx.utils import json_utils
+
+from tfx_addons.sampler.executor import Executor
+from tfx_addons.sampler.spec import SamplerSpec
+
+
+class Sampler(base_beam_component.BaseBeamComponent):
+  """A TFX component to sample examples.
+
+  The sampling component wraps an Apache Beam pipeline to process
+  data in an TFX pipeline. This component loads in tf.Record files from
+  an earlier example artifact, processes the 'train' split by default,
+  samples the split by a given label's classes, and stores the new
+  set of sampled examples into its own example artifact in
+  tf.Record format.
+
+  By default, the component will ignore all examples with a null value
```

Comment:
Can you explain the sampling logic in more details? How are examples selected to be removed from the dataset in undersampling and oversampling cases?

Reply:
Just added more info about the logic. "The sampling logic uses Python's random module: undersampling uses random.sample, and oversampling uses random.choices. Support for more complex sampling algorithms may be added at a later date."